Source/DafnyTests/DafnyTestCase.cs

```diff
@@ -0,0 +1,417 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using Microsoft.Extensions.FileProviders;
+using Xunit;
+using Xunit.Abstractions;
+using Xunit.Sdk;
+using XUnitExtensions;
+using YamlDotNet.Core;
+using YamlDotNet.Serialization;
+using YamlDotNet.Serialization.NamingConventions;
+using YamlDotNet.Serialization.ObjectFactories;
+
+namespace DafnyTests {
+  
+  public class DafnyTestSpec : IEnumerable<DafnyTestCase> {
+
+    private static DirectoryInfo OUTPUT_ROOT = new DirectoryInfo(Directory.GetCurrentDirectory());
+    
+    // TODO-RS: This is an ugly method of locating the project root.
```

Comment:
Flagging TODO (I flag TODO's to make sure they're either addressed or we have a separate issue to fix them in the future).